ci/environment.yml

```diff
@@ -7,7 +7,8 @@ dependencies:
   - toml
   - pytest
   - jupyter
-  - xeus=1.0.1
+  - xeus=5.1.0
+  - xeus-zmq=3.0.0
   - xtl
   - nlohmann_json
   - cppzmq
```

Comment:
Migration to `xeus` 5 and above allows us to drop the dependency on `xtl` and `cppzmq` for all downstream projects (mostly kernels) so you have remove them and also we should pin `nlohmann_json` to `3.11.3` wherever possible